httpcore/_async/http11.py

```diff
@@ -142,6 +138,29 @@ async def handle_async_request(self, request: Request) -> Response:
                     await self._response_closed()
             raise exc
 
+    async def _update_state(self) -> None:
+        async with self._state_lock:
+            # If the HTTP connection is idle but the socket is readable, then the
+            # only valid state is that the socket is about to return b"", indicating
+            # a server-initiated disconnect.
+            server_disconnected = (
+                self._state == HTTPConnectionState.IDLE
+                and self._network_stream.get_extra_info("is_readable")
+            )
```

Comment:
Should this check be also done in `AsyncHTTP2Connection` similarly as here?

Reply:
I think yes, but cc @tomchristie who knows better about http2